### DISCUSSION TITLE
Preprocessing for dual task

### DISCUSSION THREAD

**Discussion by fsikora** [ds4dm/ml4co-competition#78] [Q&A]
Hi all,

is there a way to avoid the preprocessing step?
We think that there will be less or none structure in the preprocessed instance, while with the original formulation, some ML model could learn more.

Best,
Florian

**Comment by gasse** [comment]
Hi @fsikora ,

I suppose you refer to the primal or the dual task ? Then no, there is no way to avoid the preprocessing step, we impose the same setup to all participants for fairness.

Still, SCIP retains a lot of information about the original problem, and has routines which allow to map some concepts from the original problem to the transformed problem, and vice-versa. See, for example, the [SCIPvarGetOrigvarSum()
](https://www.scipopt.org/doc/html/group__PublicVariableMethods.php#ga43bc31c19096e1f77959dab2b594fec1) routine. I am not really an expert in that area though, but maybe @ambros-gleixner or @antoniach would have better insights ?

Best,
Maxime

**Reply by ambros-gleixner** [reply]
Yes, and there are also further methods to obtain the variables and constraints of the original problem, see methods SCIPgetOrig* in https://www.scipopt.org/doc/html/group__GlobalProblemMethods.php#ga065118efb691d324dd06e82d34facf02

However, not all seems accessible through PySCIPOpt currently: constraint data not, also the mentioned SCIPvarGetOrigvarSum() I think; but var data is, via getVars(false).

**Reply by gasse** [reply]
If you need those methods then a quick solution is to fork the PySCIPOpt repo, then add the missing methods yourself, and use this modified pyscipopt in your agent's environment (via your `conda.yml` file). This is what we had done in a previous work, see
https://github.com/ds4dm/PySCIPOpt/tree/ml-branching

Such a modified PySCIPOpt is then easily installable in a few lines, e.g.,
```
conda install cython
pip install git+https://github.com/ds4dm/PySCIPOpt.git@ml-branching
```

Best,
Maxime
